app/services/TripJoinRequestStageManager.rb

```diff
@@ -1,38 +1,58 @@
+require "./app/services/InvalidStageChange"
+
 class TripJoinRequestStageManager
   REQUESTER = 1
-  def self.accept!(req)
-    if req.requested?
-      trip = req.trip
-      seats = trip.available_seats - (req.companions + REQUESTER)
-      trip.update_column(:available_seats, seats)
+  def self.accept!(trip_request)
+    if trip_request.requested?
+      trip = trip_request.trip
+      new_seats = trip.available_seats - (trip_request.companions + REQUESTER)
+      trip.update_column(:available_seats, new_seats)
+      new_pets = trip.pets - trip_request.pets
+      trip.update_column(:pets, new_pets)
```

Comment:
not too bad but you can make it a bit more SRP. maybe you can pass to a trip method something like 
```
things_that_are_going_to_be_used = {seats: 2, pets, 3}

class TripJoinRequestStageManager
 def self.accept!(trip_request)
    if trip_request.requested?
      trip_request.trip.update_availability({seats: (trip_request.companions + REQUESTER) , pets, trip_request.pets})
    end
end

class Trip

def update_availability(incoming_passengers)
    remaining_available_seats = available_seats -  incoming_passengers[:seats]
    #same for pets
   self.update(available_seats:   remaining_available_seats, pets: incoming_passengers[:pets)])
end
```

Reply:
If I do this, it would not be DRY anymore, I would have to rewrite some variables again just to save a couple of lines. Those three variables are used in other lines...

Reply:
that's true but right now it violates the SRP, furthermore, it creates a coupling between `Trip` and `TripJoinRequestStageManager`